drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -0,0 +1,584 @@
+				&bq274xx->stdby_current);
+		if (status < 0) {
+			LOG_ERR("Failed to read standby current");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_MAX_LOAD_CURRENT:
+		status = bq274xx_command_reg_read(bq274xx, BQ274XX_COMMAND_MAX_CURRENT,
+				&bq274xx->max_load_current);
+		if (status < 0) {
+			LOG_ERR("Failed to read maximum current");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_CHARGE:
+		status = bq274xx_command_reg_read(bq274xx, BQ274XX_COMMAND_SOC,
+				&bq274xx->state_of_charge);
+		if (status < 0) {
+			LOG_ERR("Failed to read state of charge");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY:
+		status = bq274xx_command_reg_read(bq274xx, BQ274XX_COMMAND_FULL_CAPACITY,
+				&bq274xx->full_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read full charge capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		status = bq274xx_command_reg_read(bq274xx, BQ274XX_COMMAND_REM_CAPACITY,
+				&bq274xx->remaining_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read battery remaining charge capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		status = bq274xx_command_reg_read(bq274xx, BQ274XX_COMMAND_NOM_CAPACITY,
+				&bq274xx->nom_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read battery nominal available capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY:
+		status = bq274xx_command_reg_read(bq274xx, BQ274XX_COMMAND_AVAIL_CAPACITY,
+				&bq274xx->full_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read battery full available capacity");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_AVG_POWER:
+		status = bq274xx_command_reg_read(bq274xx, BQ274XX_COMMAND_AVG_POWER,
+				&bq274xx->avg_power);
+		if (status < 0) {
+			LOG_ERR("Failed to read battery average power");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_HEALTH:
+		status = bq274xx_command_reg_read(bq274xx, BQ274XX_COMMAND_SOH,
+				&bq274xx->state_of_health);
+
+		bq274xx->state_of_health = (bq274xx->state_of_health) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read state of health");
+			return -EIO;
+		}
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+
+/**
+ * @brief initialise the fuel gauge
+ *
+ * @return 0 for success
+ */
+static int bq274xx_gauge_init(struct device *dev)
+{
+	struct bq274xx_data *bq274xx = dev->driver_data;
+	int status = 0;
+	u8_t tmp_checksum = 0, checksum_old = 0, checksum_new = 0;
+	u16_t flags = 0, designenergy_mwh = 0, taperrate = 0, id;
+	u8_t designcap_msb, designcap_lsb, designenergy_msb, designenergy_lsb,
+	     terminatevolt_msb, terminatevolt_lsb, taperrate_msb, taperrate_lsb;
+	u8_t block[32];
+
+	designenergy_mwh = (u16_t) 3.7 * DT_INST_0_TI_BQ274XX_DESIGN_CAPACITY;
```

Comment:
New DT macros should be used, e.g. in this case `DT_INST_PROP(0, design_capacity)`

Reply:
Replaced, thanks.